vsts/node-release.yaml

```diff
@@ -0,0 +1,185 @@
+name: Release_$(BuildID)_$(Date:yyyyMMdd)$(Rev:.r)
+resources:
+- repo: self
+  clean: true
+
+phases:
+
+- phase: Phase_1
+  displayName: Create Release Branch
+
+  condition: succeededOrFailed()
+  queue:
+    name: Hosted
+
+  steps:
+
+  - checkout: self  # self represents the repo where the initial Pipelines YAML file was found
+    persistCredentials: true  # set to 'true' to leave the OAuth token in the Git config after the initial fetch
+
+  - task: NodeTool@0
+    displayName: 'Use Node 6.x'
+    enabled: false
+
+  - task: Npm@1
+    inputs:
+      command: 'install'
+      workingDir: $(Build.SourcesDirectory)\.scripts\version
+    enabled: true
+
+  - powershell:  |
+      $configFile = "$(Build.SourcesDirectory)\.scripts\version\config.json"
+      $app = "$(Build.SourcesDirectory)\.scripts\version\app.js"
+
+      if ($(Test-Path -Path $configFile -PathType Leaf) -ne $true) {
+          throw "File not found: $($configFile)"
+      }
+
+      if ($(Test-Path -Path $app -PathType Leaf) -ne $true) {
+          throw "File not found: $($app)"
+      }
+
+      node $app $configFile "$(Build.SourcesDirectory)"
+      
+    displayName: Bump Version
+    enabled: true
+
+  - script: dir /b /s
+
+  - powershell: |
+      $releaseTag = Get-Date -UFormat "%Y-%m-%d"
+      $releaseBranch = "release_$($releaseTag)"
+
+      Set-Location "$(Agent.BuildDirectory)\s"
+
+      git config user.email $(GitHubEmail)
+      git config user.name $(GitHubName) 
+      git checkout -b $releaseBranch
+      git add .
+      git commit -m "Bump package versions for release $($releaseTag)"
+      git tag -d $($releaseTag)
```

Comment:
>git tag -d $($releaseTag) [](start = 6, length = 25)

this deletes the tag locally but not on the remote. if this tag has already been created and pushed, it only delays the failure. 
if we want to build in robustness and the possibility to overwrite a previous release tentative, we need to:
1. detect if the tag already exists
2. delete it locally
3. detect if the tag has been pushed
4. if yes, delete it from the remote (using git push <remote> --delete <tag name>

Reply:
I added logic to detect if the tag name already exists in local refs/tags and if not, the new tag is created

---
In reply to: [265820080](https://github.com/Azure/azure-iot-sdk-node/pull/510#discussion_r265820080) [](ancestors = 265820080)